spec/controllers/api/v1/punches_controller_spec.rb

```diff
@@ -0,0 +1,155 @@
+require 'spec_helper'
+
+describe Api::V1::PunchesController, :type => :controller do
+  let(:company) { create(:company) }
+  let(:user) { create(:user, :with_token, company: company) }
+  let(:project) { create(:project, company: user.company) }
+  let(:headers) { { token: user.token } }
+
+  describe 'POST api/v1/punches' do
+    context 'when api token is valid and params are valid' do
+      let(:params) {
+        {
+          "_json": [
+            {
+              "from": "2022-04-19T12:00:00.000Z",
+              "to": "2022-04-19T15:00:00.000Z",
+              "project_id": "#{project.id}"
+            },
+            {
+              "from": "2022-04-19T16:00:00.000Z",
+              "to": "2022-04-19T21:00:00.000Z",
+              "project_id": "#{project.id}"
```

Comment:
You don't need to use concatenation format here, you can just write `project.id`

Reply:
I will remove, thx